app/scripts/directives/loading.dir.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+angular.module('yourStlCourts').directive('loading', function($timeout){
+  return{
+    restrict: 'E',
+    scope:{
+    },
+    templateUrl: 'views/loading.html',
+    link: function($scope,element,attrs){
+      var fastEnough = true;
+      $(element).addClass("hidden");
+      $scope.$on('$stateChangeStart',function(){
+        fastEnough = false;
```

Comment:
I don't think that you can trigger multiple simultaneous state changes.  And for sure in our application, we are only ever going from one state to another. I added a $stateChangeError handler to remove the Spinner if the state change should fail.